cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,232 @@
+
+## Key scenarios
+<!-- List ~3-7 high level scenarios to clarify the value and point to how we will decompose this big area into component capabilities. We may ultimately have more than one level of scenario. -->
+The key scenarios for UDTs are outlined [here](/architecture/2024-06-resource-extensibility-feature-spec.md). We will decompose these scenarios specific to the CLI experience.
+
+### Scenario 1: Deb defines and deploys the internal messaging service using the CLI
+<!-- One or two sentence summary -->
+Deb, a platform engineer who wants to define and deploy their internal messaging service Plaid as a resource type in Radius. 
+
+Before we start designing the experience, we need to understand the high level terminologies/concept that a user would interact with in the CLI experience.
+
+| Terminology | Description | Example | 
+|-------------|-------------|---------|
+|Resource provider | Refers to a group of resource types under a common namespace. Resource providers are also entities that implement the group of resource types | Mycompany.Messaging or Applications.Core |
+|Resource type | A service or technology that can be modeled as a resource | PostgreSQL or AWS S3 or Internal messaging service |
+|Resource | An instance of a resource type | mydb or mybucket |
+
+#### User Experience
+Below is an illustrative example of how Deb would define the internal messaging service Plaid as a resource type in Radius. 
+
+1. Deb creates the resource provider and the resource type with Radius
+
+    ```bash
+    rad resource-provider create -f Mycompany.Messaging.yaml
+    ```
+    `Mycompany.Messaging.yaml` is the schema definition file that contains the resource type definition for the messaging service Plaid. See the feature spec that covers the [authoring experience for the schema definition](https://github.com/radius-project/design-notes/blob/67dd03056d22bcae2ee9743ffb308c8ac7f012bd/architecture/2024-10-user-defined-types-authoring-experience-feature-spec.md)
+
+2. Deb creates the resource instance for the messaging service Plaid
+
+    ```bash
+    rad resource create Mycompany.Messaging/Plaid myplaid -f myplaid.json
+    ```
+    The resource `myplaid` of type `Mycompany.Messaging/plaidResource`is created and available for use.
+
+4. Deb lists the resource providers and resource types available in Radius
+
+    ```bash
+    rad resource-provider list
+    ```
+    The list of resource providers are displayed.
+
+    ```bash
+    rad resource-type list --resource-provider Mycompany.Messaging
+    ```
+    The list of resource types under the resource provider `Mycompany.Messaging` are displayed.
+
+5. Deb makes changes to the resource type definition of `plaidResource` and adds to the resource provider
+
+    ```bash
+    rad resource-provider create -f Mycompany.Messaging.yaml
+    ```
+
+6. Deb wants to know the details of resource provider
+
+    ```bash
+    rad resource-provider show Mycompany.Messaging
```

Comment:
Couple of questions -
1. What's the motivation to make RP name a positional arg here but a flag in the list command?
2. What does the output here include in addition to what's displayed in the list command output?